### DISCUSSION TITLE
How to remove delay from annotation label tooltip

### DISCUSSION THREAD

**Discussion by gapetto** [reactivemarkets/react-financial-charts#568] [General]
I'm having a hard time figuring out how to customize this lib, for example I need to remove the delay from annotation label tooltip. How can I do that? I guess I need to fork the lib. Is there a short answer to the question how to add custom low-level d3 options like that for different elements when forking?

**Comment by markmcdowell** [comment]
I'm not sure what you mean by delay here. You can create your own components to add to the chart without forking anything see `https://github.com/react-financial/react-financial-charts/blob/main/packages/annotations/src/LabelAnnotation.tsx` for the LabelAnnotation implementation if that's the one you mean.

**Reply by gapetto** [reply]
Thank you for the quick reply. This is what I mean by the delay:

![Peek 2021-08-16 16-06](https://user-images.githubusercontent.com/24698150/129576667-7515cc24-2383-40f7-8d01-a5dea5962b5b.gif)

**Reply by markmcdowell** [reply]
Is that using the SvgPathAnnotation? if it is then the delay isn't customisable since it's just an svg.

You'd have to create a new Component to use instead of the path annotation that used d3 to generate the tooltip.

**Reply by gapetto** [reply]
This is original LabelAnnotation component from the lib. It has a tooltip (of type string | ((datum: any) => string)) option as one of its props.

**Reply by markmcdowell** [reply]
I see, yeah its similar for LabelAnnotation https://github.com/react-financial/react-financial-charts/blob/main/packages/annotations/src/LabelAnnotation.tsx 

It's a quite simple annotation, you can create your own and use that.

**Reply by gapetto** [reply]
Ok, I'll try that, thank you.

**Reply by gapetto** [reply]
Ok, I managed to add a custom one by creating another component. So the original "tooltip" is just an html accessibility tooltip.
